### PR TITLE
Add Tryorama tests to verify validation cases

### PR DESCRIPTION
**Pull request by jost-s**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,9 +1,8 @@
 [workspace]
-resolver = "2"
```
Comment (steveej) [inline Cargo.toml]: why was this removed?

File: build-dna.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
```
Comment (steveej) [inline build-dna.sh]:
```suggestion
#!/bin/sh
```

i can't see any bash-only features here

File: dna/zomes/integrity/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+// link_types! macro generates the enum below and impls to go in the opposite direction
+#[hdk_link_types]
+pub enum LinkTypes {
+    Fish,
+    Dog,
+    Cow,
+}
+
+impl From<LinkType> for LinkTypes {
+    fn from(x: LinkType) -> Self {
```
Comment (steveej) [inline dna/zomes/integrity/src/lib.rs]: instead of this we can use https://crates.io/crates/num_enum

**Review by steveej — COMMENTED** [review]
i can't run the tests successfully here:

```
$ nix-shell --pure --run "./build-dna.sh && npm install && npm run test"

(...)

> node --loader ts-node/esm validate.ts

(node:50496) ExperimentalWarning: --experimental-loader is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
TAP version 13
# Validate store entry - creating an entry with valid text should succeed
not ok 1 [object Object]
  ---
    operator: fail
    stack: |-
      Error: [object Object]
          at Test.assert [as _assert] (/home/steveej/src/holo/integrity-template/node_modules/tape/lib/test.js:314:54)
          at Test.bound [as _assert] (/home/steveej/src/holo/integrity-template/node_modules/tape/lib/test.js:99:32)
          at Test.fail (/home/steveej/src/holo/integrity-template/node_modules/tape/lib/test.js:408:10)
          at Test.bound [as fail] (/home/steveej/src/holo/integrity-template/node_modules/tape/lib/test.js:99:32)
          at onError (/home/steveej/src/holo/integrity-template/node_modules/tape/lib/test.js:133:22)
          at processTicksAndRejections (node:internal/process/task_queues:96:5)
  ...
# Validate store entry - creating an entry with invalid text should fail
not ok 2 should be truthy
  ---
    operator: ok
    expected: true
    actual:   false
    stack: |-
      Error: should be truthy
          at Test.assert [as _assert] (/home/steveej/src/holo/integrity-template/node_modules/tape/lib/test.js:314:54)
          at Test.bound [as _assert] (/home/steveej/src/holo/integrity-template/node_modules/tape/lib/test.js:99:32)
          at Test.assert (/home/steveej/src/holo/integrity-template/node_modules/tape/lib/test.js:433:10)
          at Test.bound [as ok] (/home/steveej/src/holo/integrity-template/node_modules/tape/lib/test.js:99:32)
          at Test.<anonymous> (file:///home/steveej/src/holo/integrity-template/validate.ts:28:11)
          at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

File: Cargo.toml (hunk 1 above)
Comment (jost-s) [reply Cargo.toml]: some rust analyzer error in the beginning, I removed this and then forgot to put it back in

**Comment by jost-s** [comment]
please run the tests again

File: dna/zomes/integrity/src/lib.rs (hunk 3 above)
Comment (jost-s) [reply dna/zomes/integrity/src/lib.rs]: okay
in fact I don't know what Eric wrote that for

File: build-dna.sh (hunk 2 above)
Comment (steveej) [reply build-dna.sh]: my previous suggestion seems to have not made it through :eyes:

**Review by steveej — COMMENTED** [review]
> please run the tests again

tests work now! however unsurprisingly so with the generous commenting out :wink:

maybe a bit meta, but what are we striving for exactly with this template?

**Comment by jost-s** [comment]
> tests work now! however unsurprisingly so with the generous commenting out 😉

That's twice the same match arm, with variations. Getting the entry type from the record is too complicated for me right now, so I didn't spend more time on fixing it.

> maybe a bit meta, but what are we striving for exactly with this template?

It serves as a template for hApp devs to copy for writing validation. That's all I know about it.

**Comment by steveej** [comment]
FYI @dukejones @zippy an updated version of this was merged in #3

**Comment by dukedorje** [comment]
Hmm, @steveeJ I just took another look at #3 and I don't see any tryorama tests -- am I missing something?

**Comment by dukedorje** [comment]
Oh I see it now -- it's in validate.ts -- yeah, already in there! I'll close this
